src/Unsafe/Coerce.purs

```diff
@@ -8,20 +8,84 @@ module Unsafe.Coerce
 -- | (that is, the caller's) responsibility to ensure that the underlying
 -- | representation for both types is the same.
 -- |
--- | One application for this function is to avoid doing work that you know is a
--- | no-op because of newtypes. For example, if you have an `Array (Conj a)` and you
--- | want an `Array (Disj a)`, you could do `Data.Array.map (runConj >>> Disj)`, but
--- | this performs an unnecessary traversal. `unsafeCoerce` accomplishes the same
--- | for free.
--- |
--- | It is highly recommended to define specializations of this function rather than
--- | using it as-is. For example:
--- |
--- | ```purescript
--- | mapConjToDisj :: forall a. Array (Conj a) -> Array (Disj a)
--- | mapConjToDisj = unsafeCoerce
+-- | Note: because this function is extraordinarily flexible, type inference
+-- | can greatly suffer. Whenever you use it, you should add type annotations
+-- | to clarify what the type of it's input value is and the type of it's
+-- | output value.
+-- |
+-- | After the v0.14.0 PureScript release, some of what was accomplished via
+-- | `unsafeCoerce` can now be accomplished via `coerce` from
+-- | `purescript-safe-coerce`. See that library's documentation for more
+-- | context.
+-- |
+-- | There are at least two ways you might want to use this function:
+-- | 1. To model existentials
+-- | 2. To model the kinds of arguments an FFI function can take if one or
+-- |    more of its arguments can be multiple types.
+-- |
+-- | ## Modeling Existentials
+-- |
+-- | Let's say you have a type you wish to use internally and expose publicly
+-- | but which end-users should not be able to view or modify. Rather, end-users
+-- | can only pass that value around in their code. One way to accomplish this
+-- | is through existentials. Existentials is not always the best solution to
```

Comment:
What you've described here isn't existentials, it's just hiding the representation of a data type. You don't need unsafeCoerce for that, you can just define the data type the normal way and avoid exporting its constructors.

Reply:
> What you've described here isn't existentials....

Ah... true. I think I had something more like Halogen Hooks in mind, which uses this pattern to create a heterogeneous array, when I wrote this example. Nate's ShowBox code is a real example of an existential.

On another note, since existentials are usually an anti-pattern, should we cover it here at all?